### PR TITLE
GCS: Make md5Hash and crc32c optional

### PR DESCRIPTION
**Pull request by barbasa**
The md5Hash and crc32c fields are still mandatory when deserializing
the Google StorageObject. An attempt to make them optional was done in
a previous PR [1].

However, when not defined a "spray.json.DeserializationException" will be thrown.

Make them both optional consistently with most of the other
StorageObjectWriteableJson parameters.

[1]: https://github.com/akka/alpakka/pull/2388

References #xxxx

### CODE REVIEW

File: google-cloud-storage/src/test/scala/akka/stream/alpakka/googlecloud/storage/scaladsl/GCStorageWiremockBase.scala [hunk 1]
```diff
@@ -76,7 +76,10 @@ abstract class GCStorageWiremockBase(_system: ActorSystem, _wireMockServer: Hove
        |  "customerEncryption": {"encryptionAlgorithm": "AES256", "keySha256": "encryption-key-sha256"},
        |  "owner": {"entity": "project-owners-123412341234", "entityId": "790607247"},
        |  "acl": [{ "kind": "storage#objectAccessControl", "id": "my-bucket/test-acl/1463505795940000", "selfLink": "https://www.googleapis.com/storage/v1/b/my-bucket/o/test-acl", "bucket": "my-bucket", "object": "test-acl", "generation": "1463505795940000", "entity": "some-entity", "role": "OWNER", "email": "owner@google.com", "entityId": "790607247", "domain": "my-domain", "projectTeam": { "projectNumber": "57959400", "team": "management" }, "etag": "R2OZrfQiij=" }]
-       |}""".stripMargin
+       |  ${maybeMd5Hash.fold("")(hash => s""","md5Hash":"$hash"""")}
```
Comment (christophermiliotis) [inline google-cloud-storage/src/test/scala/akka/stream/alpakka/googlecloud/storage/scaladsl/GCStorageWiremockBase.scala:79]:
```suggestion
       |  ${maybeMd5Hash.fold("")(hash => s""","md5Hash":"$hash"""")},
```

Also missing comma at end of L78

File: google-cloud-storage/src/test/scala/docs/scaladsl/GCStorageSourceSpec.scala [hunk 2]
```diff
@@ -296,24 +296,19 @@ class GCStorageSourceSpec
...
-
-      val getObjectGenerationSource: Source[Option[StorageObject], NotUsed] =
-        GCStorage.getObject(bucketName, fileName, Some(generation))
-
-      //#objectMetadata
+    }
 
-      val result = getObjectSource.runWith(Sink.head).futureValue
-      val resultGeneration = getObjectGenerationSource.runWith(Sink.head).futureValue
+    "get existing storage object when md5Hash and crc32c are not defined" in {
```
Comment (christophermiliotis) [inline google-cloud-storage/src/test/scala/docs/scaladsl/GCStorageSourceSpec.scala:303]:
```suggestion
    "get existing storage object when md5Hash and crc32c are not defined (composite objects)" in {
```

**Review by christophermiliotis — COMMENTED** [review]
Nice work @barbasa 👍

File: google-cloud-storage/src/test/scala/akka/stream/alpakka/googlecloud/storage/scaladsl/GCStorageWiremockBase.scala (hunk 1 above)
Comment (barbasa) [reply google-cloud-storage/src/test/scala/akka/stream/alpakka/googlecloud/storage/scaladsl/GCStorageWiremockBase.scala:79]: The comma is not needed. I have added it at the beginning of the string interpolation when the optional parameters are defined.

**Review by ennru — APPROVED** [review]
LGTM.

**Pull request merged**